astropy/cosmology/flrw/tests/test_w0cdm.py

```diff
@@ -152,13 +155,16 @@ def setup_class(self):
         self.cls = FlatwCDM
         self.cls_kwargs.update(w0=-0.5)
 
-    def test_repr(self, cosmo):
+    def test_repr(self, cosmo_cls, cosmo):
         """Test method ``.__repr__()``."""
-        assert repr(cosmo) == (
-            "FlatwCDM(name='ABCMeta', H0=<Quantity 70. km / (Mpc s)>, Om0=0.27,"
-            " w0=-0.5, Tcmb0=<Quantity 3. K>, Neff=3.04,"
-            " m_nu=<Quantity [0., 0., 0.] eV>, Ob0=0.03)"
+        super().test_repr(cosmo_cls, cosmo)
+
+        expected = (
+            "FlatwCDM(name='ABCMeta', H0=<Quantity 70. km / (Mpc s)>, Om0=0.27, "
+            "Tcmb0=<Quantity 3. K>, Neff=3.04, m_nu=<Quantity [0., 0., 0.] eV>, "
+            "Ob0=0.03, w0=-0.5)"
         )
+        assert repr(cosmo) == expected
```

Comment:
What is causing the order of some of the parameters to change and could it be avoided?

Reply:
Unfortunately not. Dataclasses strictly order the fields. And this is reflected in the string representation. The disadvantage is that it can change the representation, like here, and cannot be re-ordered. The advantage is that it is entirely deterministic and predictable. The previous repr PR brought `Cosmology` into close agreement with how dataclass repr looks, but I don't reorder the parameters, saving that for this PR.